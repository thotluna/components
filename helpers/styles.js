import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
