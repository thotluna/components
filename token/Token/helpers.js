import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
