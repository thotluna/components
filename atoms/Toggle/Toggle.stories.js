import { Toggle, styles } from '.'
import { getTemplate } from '../../helpers/storybook'

const Template = getTemplate(Toggle, styles)

export default {
  title: 'Atoms/Toggle',
  component: Toggle,
  args: {
    name: 'toggle',
    checked: false,
    onChange: () => {},
  },
  argTypes: {},
}

const args = {
  name: 'toggle',
  checked: false,
  onChange: () => {},
}

export const Default = Template.bind({})
Default.args = { ...args }

export const Active = Template.bind({})
Active.args = { ...args, isChecked: true }

export const Inactive = Template.bind({})
Inactive.args = { ...args, isChecked: false }
