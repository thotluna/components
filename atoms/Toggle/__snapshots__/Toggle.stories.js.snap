// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Toggle Active 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    aria-label="toggle"
    aria-pressed={false}
    checked={false}
    className="toggle"
    onChange={[Function]}
    role="button"
    type="checkbox"
  />
</div>
`;

exports[`Storyshots Atoms/Toggle Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    aria-label="toggle"
    aria-pressed={false}
    checked={false}
    className="toggle"
    onChange={[Function]}
    role="button"
    type="checkbox"
  />
</div>
`;

exports[`Storyshots Atoms/Toggle Inactive 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <input
    aria-label="toggle"
    aria-pressed={false}
    checked={false}
    className="toggle"
    onChange={[Function]}
    role="button"
    type="checkbox"
  />
</div>
`;
